x/evm/ante/preprocess.go

```diff
@@ -136,7 +136,9 @@ func Preprocess(ctx sdk.Context, msgEVMTransaction *evmtypes.MsgEVMTransaction)
 	if atx, ok := txData.(*ethtx.AssociateTx); ok {
 		V, R, S := atx.GetRawSignatureValues()
 		V = new(big.Int).Add(V, big.NewInt(27))
-		evmAddr, seiAddr, pubkey, err := getAddresses(V, R, S, common.Hash{}) // associate tx should sign over an empty hash
+		// Hash custom message passed in
+		customMessageHash := crypto.Keccak256Hash([]byte(atx.CustomMessage))
```

Comment:
This assumes that the caller would sign the message after Keccak256Hash'es it. I wonder whether it'd be more flexible if we just treat `custom_message` as already "hashed" and do `getAddresses(V, R, S, common.BytesToHash([]byte(atx.CustomMessage))`. That way if the caller wants Keccak256Hash they can just set `custom_message` to be the hashed value, whereas if they want other hash (or no hash at all) they can set accordingly.

Reply:
Yea think it’s better to just assume custom message isn’t hashed @codchen